### PR TITLE
Added TCI git sample for Odata using SAP S4 Hana connector

### PR DESCRIPTION
**Pull request by kramawat-tibco**

### CODE REVIEW

File: samples/app-dev/Connectors/Miscellaneous/ODataS4hana/readme.md [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+Also in flow we have Log Message for every activity output and in last Return Activity for getting all activity output.
+
+![Sample Response](../../../import-screenshots/ODataS4Hana/45.png)
+![Sample Response](../../../import-screenshots/ODataS4Hana/46.png)
+
+### Run the application
+For running the application, first you have to push the app and then scale up the app.
+Then after sometime you can see your app goes in running state.
+
+![Sample Response](../../../import-screenshots/ODataS4Hana/47.png)
```
Comment (swagupta) [inline samples/app-dev/Connectors/Miscellaneous/ODataS4hana/readme.md:196]: there are too many screenshots... you can remove this one

File: samples/app-dev/Connectors/Miscellaneous/ODataS4hana/readme.md [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+![Sample Response](../../../import-screenshots/ODataS4Hana/48.png)
```
Comment (swagupta) [inline samples/app-dev/Connectors/Miscellaneous/ODataS4hana/readme.md:197]: remove this also and check if we remove some more which are not needed... too many screenshots make sample unreadable.

File: samples/app-dev/Connectors/Miscellaneous/ODataS4hana/readme.md [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+
+1. Ensure that SAP S4 Hana services are up and running
+  
+## Import the sample
+
+1. Download the sample json file i.e. ODataS4HanaCRUD.json.
+
+2. Create a new empty app
+
+![Create an app](../../../import-screenshots/ODataS4Hana/60.png)
```
Comment (swagupta) [inline samples/app-dev/Connectors/Miscellaneous/ODataS4hana/readme.md:24]: you can reuse common screenshots from import-screenshots parent directory
check for other common screenshots as well like for flow tester

**Review by swagupta — CHANGES_REQUESTED** [review]

File: samples/app-dev/Connectors/Miscellaneous/ODataS4hana/readme.md [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+OData app with SAP S4 Hana bascially contain 2 activities i.e. OData Query and OData Modify. OData Modify contains three operation i.e. Create, Update and Delete. 
+The main purpose of these activities are:
+a) To fetch SAP S4 Hana data using OData Query activity.
+b) To create data in SAP S4 Hana server using Create operation of OData Modify activity.
+c) To update data in SAP S4 Hana server using Update operation of OData Modify activity.
+d) To delete data in SAP S4 Hana server using Delete operation of OData Modify activity.
+
+## Prerequisites
+
+1. Ensure that SAP S4 Hana services are up and running
```
Comment (vinsharm-tibco) [inline samples/app-dev/Connectors/Miscellaneous/ODataS4hana/readme.md:16]: Please add one more pre-requisite
Make sure that user have correct Root URL of OData service (i.e., SAP S4 HANA)

File: samples/app-dev/Connectors/Miscellaneous/ODataS4hana/readme.md [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+		 
+Authorization type None 
+
+![The connection](../../../import-screenshots/ODataS4Hana/4.png)
+
+Authorization type Basic
+
+![The connection](../../../import-screenshots/ODataS4Hana/5.png)
+
+Authorization type OAuth2
```
Comment (vinsharm-tibco) [inline samples/app-dev/Connectors/Miscellaneous/ODataS4hana/readme.md:72]: You can remove this along with screenshot as this Authorization type is covered in Dataverse.

File: samples/app-dev/Connectors/Miscellaneous/ODataS4hana/readme.md [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+1. Root URL - In this field we give root URL of OData services (i.e. SAP S4 Hana, Dataverse).
+2. Username - Username to connect to SAP S4 Hana.
+3. Password - Password to connect to SAP S4 Hana.
+4. Authorization Type - You can choose the authorization type from the following options:
+
+		 	None: No Username and Password is required for aunthentication.
+		 	Basic: Basic authentication using Username and Password.
+		 	OAuth2: Access the token based client authentication by using OAuth 2.0 athentication service.
+		 
+Authorization type None
```
Comment (vinsharm-tibco) [inline samples/app-dev/Connectors/Miscellaneous/ODataS4hana/readme.md:64]: Remove this as it is not required.

File: samples/app-dev/Connectors/Miscellaneous/ODataS4hana/readme.md [hunk 7]
```diff
@@ -0,0 +1,253 @@
...
+2. When hit endpoints
+
+![Sample Response](../../../import-screenshots/ODataS4Hana/58.png)
+![Sample Response](../../../import-screenshots/ODataS4Hana/59.png)
+
+
+## Troubleshooting
+
+* If you do not see the Endpoint enabled, make sure your apps is in Running status.
+* If you see test connection failed for On-cloud connection in connection tab, then check Flogo DB Services is enabled or not and your public ip and TCI environment NAT ips are whitelisted or not.
```
Comment (vinsharm-tibco) [inline samples/app-dev/Connectors/Miscellaneous/ODataS4hana/readme.md:229]: Remove this, its for DBService and not for OData

File: samples/app-dev/Connectors/Miscellaneous/ODataS4hana/readme.md [hunk 8]
```diff
@@ -0,0 +1,253 @@
...
+* If you see test connection failed for On-premies connection in connection tab, then check Flogo DB Services is enabled or not and tibagent is started for oracle database server or not.
```
Comment (vinsharm-tibco) [inline samples/app-dev/Connectors/Miscellaneous/ODataS4hana/readme.md:230]: Remove this, its for DBService and not for OData

File: samples/app-dev/Connectors/Miscellaneous/ODataS4hana/readme.md [hunk 9]
```diff
@@ -0,0 +1,253 @@
...
+* If you are not able to fech the tables or table coloumns while typing a query, then make sure your connection should be establish successfully.
+* While executing on-premise connection app from endpoint, make sure that accesskey should be selected in Hybride connectivity option from Environment controls tab and executing from flow tester make sure you have to select 'Using On-premise services' check box from flow launch configuration pop up.
```
Comment (vinsharm-tibco) [inline samples/app-dev/Connectors/Miscellaneous/ODataS4hana/readme.md:232]: Remove this, its for DBService and not for OData

File: samples/app-dev/Connectors/Miscellaneous/ODataS4hana/readme.md [hunk 10]
```diff
@@ -0,0 +1,253 @@
...
+![Sample Response](../../../import-screenshots/ODataS4Hana/58.png)
+![Sample Response](../../../import-screenshots/ODataS4Hana/59.png)
+
+
+## Troubleshooting
+
+* If you do not see the Endpoint enabled, make sure your apps is in Running status.
+* If you see test connection failed for On-cloud connection in connection tab, then check Flogo DB Services is enabled or not and your public ip and TCI environment NAT ips are whitelisted or not.
+* If you see test connection failed for On-premies connection in connection tab, then check Flogo DB Services is enabled or not and tibagent is started for oracle database server or not.
+* If you are not able to fech the tables or table coloumns while typing a query, then make sure your connection should be establish successfully.
```
Comment (vinsharm-tibco) [inline samples/app-dev/Connectors/Miscellaneous/ODataS4hana/readme.md:231]: Remove this, its for DBService and not for OData

File: samples/app-dev/Connectors/Miscellaneous/ODataS4hana/readme.md [hunk 11]
```diff
@@ -0,0 +1,253 @@
...
+
+2. When hit endpoints
+
+![Sample Response](../../../import-screenshots/ODataS4Hana/58.png)
+![Sample Response](../../../import-screenshots/ODataS4Hana/59.png)
+
+
+## Troubleshooting
+
+* If you do not see the Endpoint enabled, make sure your apps is in Running status.
```
Comment (vinsharm-tibco) [inline samples/app-dev/Connectors/Miscellaneous/ODataS4hana/readme.md:228]: Add one more point in this 
If you see test connection failed in connection tab, then check if Username and Password are correct

**Pull request merged**